docs/source/plugin/privileges.rst

```diff
@@ -0,0 +1,169 @@
+who invokes the command.
+
+As per ``require_privilege`` you can provide an error message::
+
+    @plugin.require_bot_privilege(
+        privileges.OP, 'The bot needs +o privileges.')
+
+And you **can** use both ``require_privilege`` and ``require_bot_privilege`` on
+the same plugin callable::
+
+    @plugin.require_privilege(privileges.VOICE)
+    @plugin.require_bot_privilege(privileges.OP)
+    @plugin.require_chanmsg
+    @plugin.command('special')
+    def special_command(bot, trigger):
+        # only if the user has +v and the bot has +o (or above)
+
+This way, you will allow a less privileged user to access a command for a more
+privileged bot (this work for any combination of privileges).
+
+Restrict to user account
+------------------------
+
+Sometimes, a command should be used only by properly authenticated users
+(through IRC services/NickServ), on IRC network that provides that information
+to IRC clients. This can be required with the
+:func:`~sopel.plugin.require_account` decorator::
+
+    @plugin.require_privilege(privileges.VOICE)
+    @plugin.require_account
+    @plugin.require_chanmsg
+    @plugin.command('danger')
+    def dangerous_command(bot, trigger):
+        # only if the user has +v and has a registered account
+
+This has two consequences:
+
+1. this command cannot be used by users who are not authenticated
+2. this command cannot be used on an IRC network that doesn't allow
+   authentication or doesn't expose that information
+
+It makes your plugin safer to use and prevents the possibility to use it on
+insecure IRC networks.
+
+
+Getting user privileges in a channel
+====================================
+
+Within a :term:`plugin callable`, you can get access to a user's privileges in
+a channel to check privileges manually. For example, you could adapt the level
+of information your callable provides based on said privileges.
+
+First you need a user's nick and a channel (e.g. from the trigger parameter)
+then you can get that user's privileges through the **channel**'s
+:attr:`~sopel.tools.target.Channel.privileges` attribute::
+
+    user_privileges = channel.privileges['Nickname']
+    user_privileges = channel.privileges[trigger.nick]
+
+You can check the user's privileges manually using bitwise operators. Here
+for example, we check if the user is voiced (+v) or above::
+
+    from sopel import plugin
+
+    if user_privileges & plugin.VOICE:
+        # user is voiced
+    elif user_privileges > plugin.VOICE:
+        # not voiced, but higher privileges
+        # like plugin.HALFOP or plugin.OP
+    else:
+        # no privilege
+
+Another option is to use dedicated methods from the ``channel`` object::
+
+    if channel.is_voiced('Nickname'):
+        # user is voiced
+    elif channel.has_privileges('Nickname', plugin.VOICE):
+        # not voiced, but higher privileges
+        # like plugin.HALFOP or plugin.OP
+    else:
+        # no privilege
```

Comment:
Document the _new_ location, right?

```suggestion
    from sopel import privileges

    if user_privileges & privileges.VOICE:
        # user is voiced
    elif user_privileges > privileges.VOICE:
        # not voiced, but higher privileges
        # like privileges.HALFOP or privileges.OP
    else:
        # no privilege

Another option is to use dedicated methods from the ``channel`` object::

    if channel.is_voiced('Nickname'):
        # user is voiced
    elif channel.has_privilege('Nickname', privileges.VOICE):
        # not voiced, but higher privileges
        # like privileges.HALFOP or privileges.OP
    else:
        # no privilege
```

The most important part of this suggestion is the change to `channel.has_privilege()` btw, since the actual method name has no 's'.

Reply:
Ditto @ using `sopel.plugin`  as shortcut intended for plugin authors.